evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -914,6 +942,21 @@ where
                     failure_message.to_string().encode().into(),
                 )
             }
+
+            // Check if there are any expected reverts by address that haven't been matched
+            self.expected_reverts_with_address.retain(|expected_revert| !expected_revert.found);
+
+            // If `expected_reverts_with_address` is not empty then we did not match all the
+            // expected reverts with an address
+            if let Some(pending_expected_revert) = self.expected_reverts_with_address.pop_front() {
```

Comment:
See: I don't really think we need a `VecDequeue` here if we're only checking a single event. It's b etter to use an `Option` 😅  it's making the whole code a bit more complicated than it needs to be

Reply:
I'm taking only the first element of the deque because it is the first one that did not match but there could be more after the first it.